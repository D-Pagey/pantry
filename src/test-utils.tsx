import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React, { FC, ReactNode } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { FirebaseContext } from './components/ProviderFirebase';

type AllTheProvidersProps = {
    children: ReactNode;
    customContext?: any;
};

const AllTheProviders: FC<AllTheProvidersProps> = ({ children, customContext }) => (
    <FirebaseContext.Provider
        value={{
            categories: [],
            deleteFoodItem: () => () => null,
            expiringCount: 0,
            isAuthed: false,
            isCheckingAuth: false,
            fridge: [],
            signOut: () => null,
            updateFridge: () => null,
            user: {
                email: null,
                name: null
            },
            ...customContext
        }}
    >
        <MemoryRouter>{children}</MemoryRouter>
    </FirebaseContext.Provider>
);

const customRender = (ui: any, customContext?: any, options?: any) =>
    render(ui, { wrapper: () => <AllTheProviders customContext={customContext}>{ui}</AllTheProviders>, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
