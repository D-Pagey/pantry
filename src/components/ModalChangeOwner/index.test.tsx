import { render } from '../../test-utils';
import { ModalChangeOwner } from '.';
import { TenantDan, TenantHeidi, TenantJoe } from '../../fixtures';

const props = {
    currentOwner: TenantDan,
    expiryDate: new Date(),
    itemName: 'steak',
    tenants: [TenantDan, TenantJoe, TenantHeidi],
    unit: 'servings'
};

describe('ModalChangeOwner component', () => {
    it('should render', () => {
        const { container } = render(<ModalChangeOwner {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
