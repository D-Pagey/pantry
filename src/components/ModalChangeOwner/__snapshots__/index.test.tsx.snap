// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalChangeOwner component should render 1`] = `
.c10 {
  background-color: #348841;
  border: 0;
  border-radius: 5px;
  color: white;
  cursor: pointer;
  padding: 1rem;
  min-width: 68px;
}

.c11 {
  background-color: white;
  border: 1px solid #348841;
  color: #348841;
}

.c8 {
  border-radius: 50%;
  height: auto;
  min-height: 50px;
  width: 60px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  font-weight: normal;
  text-align: center;
  margin: 0 0 1.5rem;
}

.c2 {
  font-size: 1.25rem;
  margin: 0;
  text-align: center;
}

.c6 {
  font-style: italic;
  font-weight: normal;
  margin: 2rem 0 1rem;
  text-align: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 {
  cursor: pointer;
  margin: 0 1rem 0 0;
}

.c12 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  margin: 1.5rem 0 0;
}

.c3 {
  color: #348841;
}

.c4 {
  font-weight: 500;
}

.c5 {
  color: #EB5757;
}

<div
  class="c0"
  data-testid="modalChangeOwner"
>
  <h2
    class="c1"
  >
    Change Owner
  </h2>
  <p
    class="c2"
  >
    This 
    <span
      class="c3"
    >
      servings
    </span>
     of 
    <span
      class="c3"
    >
      steak
    </span>
     is
     
    <span
      class="c3 c4"
    >
      Dan Page
      's
    </span>
     and is expiring in
     
    <span
      class="c5"
      color="#EB5757"
    >
      1 day
    </span>
    .
  </p>
  <h3
    class="c6"
  >
    Click a photo to change the owner
  </h3>
  <div
    class="c7"
  >
    <img
      alt="profile"
      class="c8 c9"
      src="https://lh5.googleusercontent.com/-2Kb4LT8qzLs/AAAAAAAAAAI/AAAAAAAAAAA/AMZuucnT8gAa__5KWDwf4PR1ro78WH3YRg/photo.jpg"
    />
    <img
      alt="profile"
      class="c8 c9"
      src="https://lh3.googleusercontent.com/a-/AOh14Gi6ZcKd1ClkJqBEEP114ZJ07XWJfQLKJKL6apgFgQ"
    />
  </div>
  <button
    class="c10 c11 c12"
    type="button"
  >
    Back
  </button>
</div>
`;
